Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml

```diff
@@ -8991,6 +8991,33 @@
         <EffectArray Operation="Set" Reference="Behavior,AP_SpawnLarva,InfoArray[0].MaxCount" Value="6"/>
         <EditorCategories value="Race:Zerg"/>
     </CUpgrade>
+    <CUpgrade id="AP_DisruptorPermanentCloak">
+        <Icon value="Assets\Textures\btn-upgrade-protoss-fenix-disruptorpermanentcloak.dds"/>
+        <EditorCategories value="Race:Protoss,UpgradeType:Talents"/>
+        <AffectedUnitArray value="AP_Disruptor"/>
+    </CUpgrade>
+    <CUpgrade id="AP_DisruptorRestructuredThrusters">
+        <Icon value="Assets\Textures\btn-upgrade-protoss-alarak-stalkerphasingarmor.dds"/>
+        <EffectArray Operation="Set" Reference="Button,AP_PurificationNovaTargeted,Tooltip" Value="Button/Tooltip/AP_PurificationNovaTargetedUpgrade"/>
+        <EditorCategories value="Race:Protoss,UpgradeType:Talents"/>
+        <AffectedUnitArray value="AP_Disruptor"/>
+    </CUpgrade>
+    <CUpgrade id="AP_DisruptorRestrainedDestruction">
+        <Icon value="Assets\Textures\btn-upgrade-stetmann-banelingmanashieldefficiency.dds"/>
+        <EditorCategories value="Race:Protoss,UpgradeType:Talents"/>
+        <AffectedUnitArray value="AP_Disruptor"/>
+    </CUpgrade>
+    <CUpgrade id="AP_DisruptorPerfectedPower">
+        <Flags index="UpgradeCheat" value="0"/>
+        <EffectArray Reference="Effect,AP_PurificationNovaDamage,Amount" Value="55"/>
+        <EffectArray Operation="Subtract" Reference="Effect,AP_PurificationNovaEnemyDamage,ShieldBonus" Value="55"/>
+        <EffectArray Reference="Effect,AP_PurificationNovaEnemyDamage,Amount" Value="55"/>
+        <EffectArray Reference="Effect,AP_PurificationNovaFriendlyDamage,Amount" Value="27.5"/>
+        <EffectArray Operation="Subtract" Reference="Effect,AP_PurificationNovaDamage,ShieldBonus" Value="55"/>
+        <EffectArray Operation="Subtract" Reference="Effect,AP_PurificationNovaFriendlyDamage,ShieldBonus" Value="27.5"/>

```

Comment:
The increase in damage to friendlies when you have restrained distruction can be construed as a nerf. I think this upgrade should just not touch the friendly-fire damage.

Reply:
Originally I just wanted to keep consistency in damage, at first I didn't think this would be much of a nerf if any but after thinking about it more I can see how this would be a problem in certain scenarios. Increased friendly-fire damage removed.